rest/src/main/kotlin/builder/interaction/OptionsBuilder.kt

```diff
@@ -58,36 +65,66 @@ public sealed class BaseChoiceBuilder<T>(
     private var _choices: Optional<MutableList<Choice<*>>> = Optional.Missing()
     public var choices: MutableList<Choice<*>>? by ::_choices.delegate()
 
-    public abstract fun choice(name: String, value: T)
+    public abstract fun choice(name: String, value: T, nameLocalizations: Optional<Map<Locale, String>>)
+
+    /**
+     * Registers a new choice with [name] representing value and applies [localizationsBuilder] to it
+     *
+     * @see ChoiceLocalizationsBuilder
+     */
+    public inline fun choice(name: String, value: T, localizationsBuilder: ChoiceLocalizationsBuilder.() -> Unit = {}) {
+        val localizations = ChoiceLocalizationsBuilder(name)
+            .apply(localizationsBuilder)
+            .nameLocalizations
+            ?: emptyMap()
+        val optional = if (localizations.isEmpty()) {
+            Optional.Missing()
+        } else {
+            Optional(localizations)
+        }
+
+        return choice(name, value, optional)
```

Comment:
```suggestion
        val builder = ChoiceLocalizationsBuilder(name).apply(localizationsBuilder)
        choice(builder.name, value, builder._nameLocalizations)
```
adjust to suggestion for `ChoiceLocalizationsBuilder`